### ISSUE TITLE
Visual Studio 2017 and suspend and resume

### ISSUE THREAD

**Issue by pekspro** [Windows-XAML/Template10#1501]
I just created a new project, based on the Hamburger template, in Visual Studio 2017. If I launch the application in the debugger, selects **Suspend and shutdown**, and then launch the application it’s not resumed as expected. Instead it’s restarted. In **OnStartAsync** the value of **PreviousExecutionState** is always **NotRunning**.

I have two projects I’m working based on Template 10. Both have the same problem as above, but for some reason suspend and resume works fine in Visual Studio 2015. None Template 10 projects also seem to support suspend and resume properly in Visual Studio 2017. It’s very confusing.

**Comment by pcolmer** [comment]
I'm seeing this with the Minimal template as well. This is with the released v1.1.12. I haven't got VS2015 on my computer still so I can't verify that aspect of it.

If I upgrade T10 to v1.1.13-preview-20161207 and repeat the start/suspend & shutdown, suspending causes this exception:

"The group or resource is not in the correct state to perform the requested operation. (Exception from HRESULT: 0x8007139F)"

**Comment by JerryNixon** [comment]
Yep. Clearly PreviousExceptionState is not reliable right now.

**Comment by JerryNixon** [comment]
You know @pekspro I can't thank you enough for posting this because it's killing me and it's nice to know it's not something I am doing.

**Comment by pekspro** [comment]
@JerryNixon, I´m glad to see that a bug report could make you happy :-) I reported this as a bug in Visual Studio for a while ago:
https://developercommunity.visualstudio.com/content/problem/50470/cant-resume-uwp-application-after-suspend-and-shut.html

Since it affected all my applications I assumed it was something with Visual Studio. For a few weeks ago I tried Template Studio, and created a project and then resume/suspend worked. That was a bit unexpected! Then I created a basic Template 10 project, and the problem was back. But it worked well in Visual Studio 2015…

I'm bit fascinated by bugs like this :-) Do we know if the bug is in Template10 or VS2017? I suspect the latter, but since it works with other frameworks the evidence is not clear.

**Comment by JerryNixon** [comment]
I can now confirm that I caused it.

**Comment by JerryNixon** [comment]
The core problem is the extended session logic added to template 10 a few months ago. It actually prevents suspension making your app NOT suspend when it is minimized. On one hand it is awesome. On the other, it makes it so your app does not suspend - better said, so your app suspension cannot be tested without invoking suspend and then waiting 10 minutes for the session extension to expire. Ha! In version 10.2 I have removed this feature, actually, I have made it a setting to toggle on and off and default to off.

**Comment by JerryNixon** [comment]
I suppose that does not address the issue in Template 10.1 does it? Okay. I will go into the current branch and disable the extended session for you and publish a new Nuget for it in about a month. I say in about a month because I am trolling through the issues for critical other things that might need to be pushed into 10.1 so we don't release two or three new ones for a small batch of updates. To be honest, I am working on 10.2 all the time and I am not looking forward to opening 10.1, but I know this is a deal.

**Comment by pekspro** [comment]
Thanks @JerryNixon, that's a pretty good explanation :-). But I just tried to set AutoExtendExecutionSession to false manually but that didn't solve the problem :-( I also looked in Template Studio and ExtendedExecutionSession doesn't seem to be used in there at all. I have also tried to wait for more than 10 minutes after the application been terminated, both with Template 10 and Template Studio, and the problem remained.  So, I think there is something more going on.

That said, making a new version may be a good idea but personally I don't need it. If I could work around the problem that would be good enough for me :-)
